CONTRIBUTING.md

```diff
@@ -87,3 +87,376 @@ generated using the pre-commit hooks for
+
+    ```shell
+    kubectl create secret generic regcred \
+      --from-file=.dockerconfigjson=/var/tmp/voxel51-docker.json \
+      --type kubernetes.io/dockerconfigjson \
+      --namespace "${NAMESPACE}" \
+    ```
+
+1. In [skaffold.yaml](./skaffold.yaml)
+   set `imagePullSecrets` for the helm release named `fiftyone-teams-app`
+   in `setValueTemplates.imagePullSecrets[0].name=regcred`
+
+    ```yaml
+    deploy:
+      helm:
+        releases:
+          - name: fiftyone-teams-app
+            setValueTemplates:
+              imagePullSecrets:
+                - name: regcred
+    ```
+
+1. Run skaffold
+
+    ```shell
+    skaffold dev
+    ```
+
+For more information, see the Kubernetes documentation
+[Pull an Image from a Private Registry](https://kubernetes.io/docs/tasks/configure-pod-container/pull-image-private-registry/).
+
+> _Note:_ When `minikube delete`, the secret `regcred` must be recreated.
+
+##### Google Artifact Repository
+
+To run released images from Google Artifact repository in the
+GCP project `computer-vision-team`, configure minikube and skaffold
+
+1. Configure GCP Credentials
+   [gcloud auth](https://cloud.google.com/sdk/gcloud/reference/auth)
+1. Configure
+   [gcloud auth application-default](https://cloud.google.com/sdk/gcloud/reference/auth/application-default)
+
+1. Start minikube and enable the addon `gcp-auth`
+
+    ```shell
+    minikube start
+    minikube addons enable gcp-auth
+    ```
+
+1. In [skaffold.yaml](./skaffold.yaml)
+   comment `imagePullSecrets` for the helm release named `fiftyone-teams-app`
+   in `setValueTemplates.imagePullSecrets[0].name=regcred`
+
+    ```yaml
+    deploy:
+      helm:
+        releases:
+          - name: fiftyone-teams-app
+            setValueTemplates:
+              # imagePullSecrets:
+              #   - name: regcred
+    ```
+
+1. For each service
+
+    * `apiSettings`
+    * `appSettings`
+    * `pluginsSettings`
+    * `teamsAppSettings`
+
+   update the `image.repository` value to be `<REPOSITORY>/<IMAGE_NAME>` and
+   the corresponding `image.tag` value.  For example for the version `1.6.0` at `dev7`
+
+    ```yaml
+    apiSettings:
+      image:
+        repository: us-central1-docker.pkg.dev/computer-vision-team/dev-docker/fiftyone-teams-api
+        tag: v1.6.0.dev7
+    appSettings:
+      image:
+        repository: us-central1-docker.pkg.dev/computer-vision-team/dev-docker/fiftyone-app
+        tag: v1.6.0.dev7
+    pluginsSettings:
+      image:
+        repository: us-central1-docker.pkg.dev/computer-vision-team/dev-docker/fiftyone-app
+        tag: v1.6.0.dev7
+    teamsAppSettings:
+      image:
+        repository: us-central1-docker.pkg.dev/computer-vision-team/dev-docker/fiftyone-teams-app
+        # Note: the naming convention for the image `fiftyone-teams-app` differs from the other images `fiftyone-app`, `fiftyone-app` and `fiftyone-teams-api`
+        # the others are `v1.6.0.dev7` (not `.dev7` vs `-dev7`)
```

Comment:
I'm not sure how that relate to how we tag an image.  Can't we make the version tags consistent?

Reply:
`${APP_IMAGE_NAME//-dev/.dev}`

Works in the case of `-dev` but when we look at the sha-based builds, things get really wonkly:

In `newauth-dev-fiftyone-ai` we currently have:
`fiftyone-teams-app:v1.6.0-f64bd6e.0` and `fiftyone-teams-api:v1.6.0.dev8_f64bd6e`

That's already a mutation because PEP440 calls for `1.6.0.dev8+f64bd6e` but docker tags can't contain a `+`

In the very near future we'll start to iterate version numbers on `app` at a different rate than `api` so we can't even "just steal" the version number from api and slap it on an app image..

So... sure... we can... there's always an answer.... I just generally try to use the native tools associated with the language so people who `npm` and people who `python` aren't forced to use `tophers-version-tool-of-doom` to figure out what number they're supposed to use...

(Then we can talk about the fact that `npm version prerelease --preid $${SHORT_SHA}` outputs a version that starts with a `v` and `newversion package | newversion bump "${_NEWVERSION_BUMP}"` does not 😉 )

Two different tools with very different versioning rules, and (hopefully soon) two different repos so they're not always being built at the same time...

Reply:
a boy can dream...